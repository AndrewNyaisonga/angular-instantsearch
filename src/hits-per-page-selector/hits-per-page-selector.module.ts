import { CommonModule } from "@angular/common";
import { NgModule } from "@angular/core";

import { NgISFooterModule } from "../footer/footer.module";
import { NgISHeaderModule } from "../header/header.module";
import { NgISHitsPerPageSelector } from "./hits-per-page-selector";

@NgModule({
  declarations: [NgISHitsPerPageSelector],
  entryComponents: [NgISHitsPerPageSelector],
  exports: [NgISHitsPerPageSelector],
  imports: [CommonModule, NgISHeaderModule, NgISFooterModule]
})
export class NgISHitsPerPageSelectorModule {}
