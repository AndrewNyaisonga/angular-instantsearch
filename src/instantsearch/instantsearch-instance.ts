import { Injectable } from "@angular/core";
import instantsearch from "instantsearch.js/es";

@Injectable()
export class NgISInstance {
  public appId: string;
  public apiKey: string;
  public indexName: string;

  public numberLocale?: string;
  public searchFunction?: () => void;
  public createAlgoliaClient?: () => object;
  public searchParameters?: object | void;
  public urlSync?:
    | boolean
    | {
        mapping?: object;
        threshold?: number;
        trackedParameters?: string[];
        useHash?: boolean;
        getHistoryState?: () => object;
      };

  private instance?: {
    start: () => void;
  };

  public init(config: object) {
    this.instance = instantsearch(config);
  }

  public start() {
    this.instance.start();
  }
}
