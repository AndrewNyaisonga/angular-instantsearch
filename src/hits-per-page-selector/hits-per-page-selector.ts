import { Component, Input } from "@angular/core";
import { connectHitsPerPage } from "instantsearch.js/es/connectors";
import { noop } from "lodash";

import { BaseWidget } from "../base-widget";
import { NgAisInstance } from "../instantsearch/instantsearch-instance";
import { bem } from "../utils";

const cx = bem("HitsPerPageSelector");

@Component({
  selector: "ng-ais-hits-per-page-selector",
  template: `
    <div class="${cx()}">
      <ng-ais-header [header]="header" className="${cx(
        "header"
      )}"></ng-ais-header>

      <div class="${cx("body")}">
        <select
          class="${cx("select")}"
          (change)="state.refine($event.target.value)"
        >
          <option
            class="${cx("option")}"
            *ngFor="let item of state.items"
            [value]="item.value"
            [selected]="item.isRefined"
          >
            {{item.label}}
          </option>
        </select>
      </div>

      <ng-ais-footer [footer]="footer" className="${cx(
        "footer"
      )}"></ng-ais-footer>
    </div>
  `
})
export class NgAisHitsPerPageSelector extends BaseWidget {
  @Input()
  public items: {
    value: number;
    label: string;
    default?: boolean;
  }[];

  public state = {
    items: [],
    refine: noop
  };

  constructor(searchInstance: NgAisInstance) {
    super(searchInstance);
  }

  public ngOnInit() {
    this.createWidget(connectHitsPerPage, { items: this.items });
    super.ngOnInit();
  }
}
