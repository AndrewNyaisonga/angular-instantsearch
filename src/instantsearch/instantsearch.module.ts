import { CommonModule } from "@angular/common";
import { ModuleWithProviders, NgModule } from "@angular/core";

import { NgISInstantSearch } from "./instantsearch";
import { NgISInstance } from "./instantsearch-instance";

@NgModule({
  declarations: [NgISInstantSearch],
  entryComponents: [NgISInstantSearch],
  exports: [NgISInstantSearch],
  imports: [CommonModule]
})
export class NgISInstantSearchModule {
  public static forRoot(): ModuleWithProviders {
    return {
      ngModule: NgISInstantSearchModule,
      providers: [NgISInstance]
    };
  }
}
