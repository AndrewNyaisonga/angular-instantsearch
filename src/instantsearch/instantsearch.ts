import { AfterViewInit, Component, Input, OnInit } from "@angular/core";

import { NgISInstance } from "./instantsearch-instance";

@Component({
  selector: "ngis-instantsearch",
  template: `<ng-content></ng-content>`
})
export class NgISInstantSearch implements AfterViewInit, OnInit {
  @Input() public config: InstantSearchConfig;

  constructor(private searchInstance: NgISInstance) {}

  public ngOnInit() {
    this.searchInstance.init(this.config);
  }

  public ngAfterViewInit() {
    this.searchInstance.start();
  }
}
