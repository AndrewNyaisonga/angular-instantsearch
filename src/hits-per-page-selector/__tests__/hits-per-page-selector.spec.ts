import { TestBed } from "@angular/core/testing";

import { NgAisInstantSearchModule } from "../../instantsearch/instantsearch.module";
import { NgAisHitsPerPageSelectorModule } from "../hits-per-page-selector.module";
import { NgAisHitsPerPageSelector } from "../hits-per-page-selector";

jest.mock("../../base-widget");

describe("HitsPerPageSelector", () => {
  beforeEach(() =>
    TestBed.configureTestingModule({
      declarations: [],
      imports: [
        NgAisInstantSearchModule.forRoot(),
        NgAisHitsPerPageSelectorModule
      ]
    })
  );

  it("renders markup without state", () => {
    const fixture = TestBed.createComponent(NgAisHitsPerPageSelector);
    expect(fixture).toMatchSnapshot();
  });

  it("renders markup with state", () => {
    const fixture = TestBed.createComponent(NgAisHitsPerPageSelector);

    fixture.componentInstance.updateState(
      {
        items: [
          { value: 10, label: "10 per page", isRefined: true },
          { value: 20, label: "20 per page" },
          { value: 30, label: "30 per page" }
        ],
        refine: jest.fn()
      },
      false
    );
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it("should refine() when selecting another option", () => {
    const refine = jest.fn();
    const fixture = TestBed.createComponent(NgAisHitsPerPageSelector);

    fixture.componentInstance.updateState(
      {
        refine,
        items: [
          { value: 10, label: "10 per page", isRefined: true },
          { value: 20, label: "20 per page" },
          { value: 30, label: "30 per page" }
        ]
      },
      false
    );
    fixture.detectChanges();

    const select = fixture.debugElement.nativeElement.querySelector("select");
    select.value = "20";
    select.dispatchEvent(new Event("change"));

    expect(refine).toHaveBeenCalled();
    expect(refine).toHaveBeenCalledWith("20");
  });
});
