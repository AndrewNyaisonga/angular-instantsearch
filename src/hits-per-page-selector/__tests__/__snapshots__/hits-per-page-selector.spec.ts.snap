// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HitsPerPageSelector renders markup with state 1`] = `
<ng-ais-hits-per-page-selector
  cx={[Function Function]}
  searchInstance={[Function NgAisInstance]}
  state={[Function Object]}
  updateState={[Function Function]}
>
  
      <div
    class="ais-HitsPerPageSelector"
  >
    
        
    <ng-ais-header
      ng-reflect-class-name="ais-HitsPerPageSelector-header"
    >
      
      
      
      
    
    </ng-ais-header>
    
  
        
    <div
      class="ais-HitsPerPageSelector-body ais-body"
    >
      
          
      <select
        class="ais-HitsPerPageSelector-select"
      >
        
            
        
        <option
          class="ais-HitsPerPageSelector-option"
          value="10"
        >
          
              10 per page
            
        </option>
        <option
          class="ais-HitsPerPageSelector-option"
          value="20"
        >
          
              20 per page
            
        </option>
        <option
          class="ais-HitsPerPageSelector-option"
          value="30"
        >
          
              30 per page
            
        </option>
        
          
      </select>
      
        
    </div>
    
  
        
    <ng-ais-footer
      ng-reflect-class-name="ais-HitsPerPageSelector-footer"
    >
      
      
      
      
    
    </ng-ais-footer>
    
      
  </div>
    
</ng-ais-hits-per-page-selector>
`;

exports[`HitsPerPageSelector renders markup without state 1`] = `
<ng-ais-hits-per-page-selector
  cx={[Function Function]}
  searchInstance={[Function NgAisInstance]}
  state={[Function Object]}
  updateState={[Function Function]}
>
  
      <div>
    
        
    <ng-ais-header>
      
      
      
      
    
    </ng-ais-header>
    
  
        
    <div>
      
          
      <select>
        
            
        
        
          
      </select>
      
        
    </div>
    
  
        
    <ng-ais-footer>
      
      
      
      
    
    </ng-ais-footer>
    
      
  </div>
    
</ng-ais-hits-per-page-selector>
`;
